### PR TITLE
CA-250785: Fix synchronisation of VGPU migration operations

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

File: lib/xenops_server.ml [hunk 1]
```diff
@@ -1597,6 +1602,7 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.t) : unit =
 			debug "VM.receive_memory %s power_state = %s" id (state.Vm.power_state |> rpc_of_power_state |> Jsonrpc.to_string);*)
 
 			(* set up the destination domain *)
+			debug "VM.receive_memory creating domain and restoring VIFs";
```
Comment (lindig) [inline lib/xenops_server.ml]: Have these log messages enough context such that we know which VM and domain we are dealing with when reading the logs? I would also ask that about the log messages below.

Comment (mseri) [inline lib/xenops_server.ml]: This should be ok, we have the events in the queue and the related task ids that allow for easy tracing of the vm.

File: lib/xenops_server.ml [hunk 2]
```diff
@@ -1560,26 +1560,31 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.t) : unit =
...
+						save ();
+						final_handshake ()
 					| [vgpu_id] ->
 						let vgpu_url = make_url "/migrate-vgpu/" (VGPU_DB.string_of_id vgpu_id) in
 						Open_uri.with_open_uri vgpu_url (fun vgpu_fd ->
 							do_request vgpu_fd [] vgpu_url;
-							save_vm_then_handshake ~vgpu:(vgpu_id, vgpu_fd) ()
-						)
+							save ~vgpu:(vgpu_id, vgpu_fd) ();
+							Handshake.recv_success vgpu_fd;
```
Comment (mseri) [inline lib/xenops_server.ml]: Aha!

File: lib/xenops_server.ml [hunk 3]
```diff
@@ -2175,16 +2182,12 @@ module VM = struct
...
-					let vm_id = VGPU_DB.vm_of vgpu_id in
-					if is_localhost then begin
-						immediate_operation dbg vm_id op;
-						None
-					end else
-						Some (queue_operation dbg vm_id op)
-				in
-				Opt.iter (fun t -> t |> Xenops_client.wait_for_task dbg |> ignore) task
+				let vm_id = VGPU_DB.vm_of vgpu_id in
+				(* This must happen in parallel with a VM_receive_memory operation, so
```
Comment (mseri) [inline lib/xenops_server.ml]: I would also add a reference to the issue in the comment.

Comment (robhoes) [inline lib/xenops_server.ml]: Sure.

**Pull request merged**